### PR TITLE
implemented all snapshot related endpoints in mock 

### PR DESCRIPTION
**Pull request by dkijania**
(update snapshot/get delegators / voters) and put new fund endpoint. Fixed typo in vit-ss project and exposed method which combines contributors and voters. Additionally improve rest module organization in mock

### CODE REVIEW

File: src/vit-testing/vitup/src/mode/mock/rest/control/handlers.rs [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+pub async fn logs_clear(context: ContextLock) -> Result<impl Reply, Rejection> {
+    let mut context_lock = context.lock().unwrap();
+    context_lock.clear_logs();
+    Ok(warp::reply())
+}
+
+pub async fn file_lister_handler(context: ContextLock) -> Result<impl Reply, Rejection> {
+    let context_lock = context.lock().unwrap();
+    Ok(dump_json(context_lock.working_dir())?).map(|r| warp::reply::json(&r))
```
Comment (cong-or) [inline src/vit-testing/vitup/src/mode/mock/rest/control/handlers.rs:23]: how is the context set? Where is the working dir defined?

File: src/vit-testing/vitup/src/mode/mock/snapshot.rs [hunk 2]
```diff
@@ -1,64 +1,161 @@
...
+    snapshot_tags: BTreeMap<String, i64>,
+    voters: Vec<Voter>,
+    contributions: Vec<Contribution>,
 }
 
 impl VoterSnapshot {
     pub fn from_config_or_default(
-        defined_wallets: Vec<(WalletAlias, &WalletSettings)>,
-        snapshot_config: &Option<SnapshotInitials>,
+        wallets: Vec<(WalletAlias, &Wallet)>,
```
Comment (cong-or) [inline src/vit-testing/vitup/src/mode/mock/snapshot.rs:21]: the Wallet is mapped to a config?

**Review by cong-or — CHANGES_REQUESTED** [review]

**Review by cong-or — APPROVED** [review]
Other than a few questions. LGTM.

File: src/voting-tools-rs/src/logic.rs [hunk 3]
```diff
@@ -13,6 +13,10 @@ use microtype::Microtype;
 use crate::model::{network_info, Delegations, Output, Reg, SlotNo, StakeVKey, TestnetMagic};
 use crate::Db;
 
+pub trait DataProvider{
```
[the remark this answers is not in the document]
Comment (dkijania) [reply src/voting-tools-rs/src/logic.rs]: Apologies, leftovers from different PR

File: src/vit-servicing-station/vit-servicing-station-lib/src/v0/endpoints/snapshot/mod.rs [hunk 4]
```diff
@@ -218,6 +205,28 @@ pub async fn update_from_shanpshot_info(
...
+}
+
+#[tracing::instrument(skip(snapshot, context))]
+pub async fn update_from_snapshot_info(
+    tag: String,
+    snapshot: impl IntoIterator<Item = SnapshotInfo>,
+    update_timestamp: i64,
+    context: SharedContext,
+) -> Result<(), HandleError> {
+    let pool = &context.read().await.db_connection_pool;
```
[the remark this answers is not in the document]
Comment (dkijania) [reply src/vit-servicing-station/vit-servicing-station-lib/src/v0/endpoints/snapshot/mod.rs:218]: TBH I've also wondered about this. But just blindly replicate this concept from other methods in vit-servicing-station

File: src/vit-testing/vitup/src/mode/mock/rest/control/handlers.rs (hunk 1 above)
Comment (dkijania) [reply src/vit-testing/vitup/src/mode/mock/rest/control/handlers.rs:23]: working dir is set in configuration. Context is set based on configuration and represent mock state (ledger/vit-ss data)

File: src/vit-testing/vitup/src/mode/mock/snapshot.rs (hunk 2 above)
Comment (dkijania) [reply src/vit-testing/vitup/src/mode/mock/snapshot.rs:21]: configuration can define wallets, yes

**Review by FelipeRosa — APPROVED** [review]

**Pull request merged**